src/Adapter/Product/Combination/Update/CombinationStockUpdater.php

```diff
@@ -138,53 +146,50 @@ private function fillUpdatableProperties(Combination $combination, CombinationSt
      */
     private function updateStockAvailable(Combination $combination, CombinationStockProperties $properties): void
     {
-        $updateQuantity = null !== $properties->getQuantity();
+        $updateQuantity = null !== $properties->getDeltaQuantity();
         $updateLocation = null !== $properties->getLocation();
 
         if (!$updateQuantity && !$updateLocation) {
             return;
         }
 
-        $newQuantity = $properties->getQuantity();
-        $newLocation = $properties->getLocation();
-
         $stockAvailable = $this->stockAvailableRepository->getForCombination(new CombinationId((int) $combination->id));
 
         if ($updateQuantity) {
-            $this->saveMovement($combination, (int) $stockAvailable->quantity, $newQuantity);
-            $stockAvailable->quantity = $newQuantity;
+            $this->updateQuantity($stockAvailable, $properties->getDeltaQuantity());
         }
 
         if ($updateLocation) {
-            $stockAvailable->location = $newLocation;
+            $stockAvailable->location = $properties->getLocation();
         }
 
         $this->stockAvailableRepository->update($stockAvailable);
     }
 
     /**
-     * @param Combination $combination
-     * @param int $oldQuantity
-     * @param int $newQuantity
+     * @param StockAvailable $stockAvailable
+     * @param int $deltaQuantity
      *
-     * @throws CoreException
+     * @todo: this method is indetical to ProductStockUpdater:updateQuantity, whats the best way to reuse it?
```

Comment:
```suggestion
     * @todo: this method is identical to ProductStockUpdater:updateQuantity, whats the best way to reuse it?
```

Is it a coincidence (= both follow the same logic, but it's 2 different processes) or is it identical because we're doing the same domain scenario ?

If we're doing the same, maybe we should create a `ProductStockUpdater::updateCombinationQuantity()` function that rely/reuse/decorate  ù ProductStockUpdater::updateQuantity()`

Reply:
we are doing the same. Updates stock movement using stock available (stock available is already aware of product/combination id). Basically it just makes sure we provide the correct movement reason to stockManager->saveMovement 
But now I also wonder, if its right to have the stock movement reason hardcoded here? in theory we should be able to use the updater service in other cases, for example - if we update stock after order cancellation, then movement reason should be another one.

Reply:
Indeed this seems to belong to a 'Stock management' domain in charge of such calculations.

Reply:
Indeed both use cases are related to modifications of the stock done by an employee They are management modifications as you say, so they have to use the appropriate stock movement defined in the Configuration (one for increase, one for decrease).

About the duplicated code, I don't think it's too important Both services (productStockUpdater and CombinationStockUpdater) have been created separately for clarity and to avoid mixing too many features from combination in the product service, they are bound to have similar code since they perform similar features.

That said if you really want to avoid duplication I guess it's doable, you could move this piece of code into the `StockAvailableRepository` for example, it's adapted for this need and both services already use it.